model-training/manifest/data_classes.py

```diff
@@ -0,0 +1,165 @@
+import os
+import json
+from typing import Dict, List
+from dataclasses import dataclass, field
+
+@dataclass
+class TrainingSample():
+    """Represents a transcription sample in The People's Speech."""
+    duration_ms: int
+    label: str
+    name: str
+
+    def to_nemo_line(self, audio_dir=None) -> str:
+        """Formats the TrainingSample into a NeMo-style manifest line.
+        
+        Args:
+            audio_dir (str): Optional parent directory in which all audio files
+            are located (leave empty if using tarred audio files)
+        Returns:
+            a NeMo-style, one-line string representation of the TrainingSample
+        """
+        if audio_dir is None:
+            audio_filepath = self.name
+        else:
+            audio_filepath = os.path.join(audio_dir, self.name)
+        as_nemo_dict = {
+            "duration": self.duration_ms / 1000,
+            "text": self.label,
+            "audio_filepath": audio_filepath
+        }
+        return json.dumps(as_nemo_dict)
+
+@dataclass
+class TrainingData():
+    """Represents the transcription samples extracted from a single
+    source audio file."""
+    samples: List[TrainingSample] = field(default_factory=list)
+
+    def append_data(self, duration_ms: int, label: str, name: str):
+        """Create and append a new TrainingSample"""
+        new_sample = TrainingSample(
+            duration_ms=duration_ms,
+            label=label,
+            name=name
+        )
+        self.samples.append(new_sample)
+
+    def get_total_duration_ms(self) -> float:
+        """Get the summed duration of the TrainingSamples in the object."""
+        total_duration_ms = 0
+        for sample in self.samples:
+            total_duration_ms += sample.duration_ms
+        return total_duration_ms
+
+    def to_dict(self) -> Dict:
```

Comment:
Are these to_dict methods ever called? Is ther a reason to keep them?

Reply:
I've used them before to facilitate reading and writting manifests (subsets, or going from NeMo to TPS format). I don't think they ended up being called in the code I pushed though. Could drop them, no big feelings on this one